File: annotation/src/main/java/com/semye/annotation/SemyeRuntime.java
package com.semye.annotation;

import java.lang.annotation.*;


@Documented
@Target({ElementType.TYPE,
        ElementType.FIELD,
        ElementType.METHOD,
        ElementType.PARAMETER
})
@Retention(RetentionPolicy.RUNTIME)
public @interface SemyeRuntime {
    String name() default "semye";

    int value() default 0;
}

File: annotation/src/main/java/com/semye/annotation/SemyeRuntime.kt
package com.semye.annotation

@MustBeDocumented
@Target(
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CLASS,
    AnnotationTarget.FIELD,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.VALUE_PARAMETER
)
@Retention(AnnotationRetention.RUNTIME)
annotation class SemyeRuntime(val name: String = "semye", val value: Int = 0)